ecommerce/api_test.py

```diff
@@ -0,0 +1,46 @@
+"""
+Test for ecommerce functions
+"""
+from django.http.response import Http404
+
+from courses.factories import CourseRunFactory
+from dashboard.factories import ProgramEnrollmentFactory
+from ecommerce.api import get_purchasable_course_run
+from ecommerce.factories import CoursePriceFactory
+from search.base import ESTestCase
+
+
+# pylint: disable=no-self-use
+class PurchasableTests(ESTestCase):
+    """
+    Tests for get_purchasable_courses
+    """
+    def setUp(self):
+        """
+        Create common objects
+        """
+        super(PurchasableTests, self).setUp()
+
+        self.course_run = CourseRunFactory.create(course__program__live=True)
+        # Also creates a user
+        self.program_enrollment = ProgramEnrollmentFactory.create(program=self.course_run.course.program)
+        self.user = self.program_enrollment.user
+        CoursePriceFactory.create(course_run=self.course_run, is_valid=True)
```

Comment:
Please move the creation of these data to the `setUpTestData`: we do not need to recreate the entries every time.

If you have some specific parts that get changed by a test, you can use `setUp` to restore the default status